packages/opentelemetry-context-async-hooks/src/AbstractAsyncHooksContextManager.ts

```diff
@@ -146,10 +146,14 @@ export abstract class AbstractAsyncHooksContextManager
     const contextManager = this;
     return function (this: never, event: string) {
       const map = contextManager._getPatchMap(ee);
-      if (map?.[event] !== undefined) {
-        delete map[event];
+      if (map !== undefined) {
```

Comment:
```
const map = null; 
map !== undefined; 

map['aa'] !== undefined; 
// error
```

```suggestion
      if (map) {
```

Reply:
I use `!== undefined` because it's used at a lot places in this file.

Reply:
but if the map is `null` then the condition will pass and you will have undefined object exception

Reply:
`!= null` would catch both null and undefined

Reply:
Also, the map will never be `null` in this case, only `undefined`. If a dev made a change which allowed this to be `null`, the test would fail.

Reply:
I agree that map is never null here, i don't see the benefit of checking for both undefined and null

Reply:
As said I usually use `!= null`. But there are a lot places in this file where `!== undefined` is already used so the no 1 rule 'When in Rome, Do as the Romans Do' wins.
I would prefer to do cleanups like this in a non bugfix PR.

Reply:
well `if (map) {` would simply do the trick and will be a bullet proof for any future changes, you don't need to use `!= null`